libr/anal/anal.c

```diff
@@ -118,17 +118,15 @@ R_API RAnal *r_anal_new() {
 	anal->plugins = r_list_newf ((RListFree) r_anal_plugin_free);
 	if (anal->plugins) {
 		for (i=0; anal_static_plugins[i]; i++) {
-			static_plugin = R_NEW (RAnalPlugin);
-			*static_plugin = *anal_static_plugins[i];
-			r_anal_add (anal, static_plugin);
+			r_anal_add (anal, anal_static_plugins[i]);
 		}
 	}
 	return anal;
 }
 
 R_API void r_anal_plugin_free (RAnalPlugin *p) {
 	if (p && p->fini) {
-		p->fini (p);
+		p->fini (NULL);
 	}
```

Comment:
if there is no ini in the declaration is NULL thus is not called but you can't pass NULL to fini because if i develop an anal plugin with a fini function will not be called as i expect. 

Regarding plugin_tms320 it doesn't use it because it has an static struct so the parameter is unused but the API is the way it is.